lib/src/intl_suggestors/intl_migrator.dart

```diff
@@ -163,6 +169,46 @@ class IntlMigrator extends ComponentUsageMigrator {
     // }
   }
 
+  // recursive function to if previous node is ignore comment until we come to the previous ";"
+  // limit
+  bool isIgnoreCommentBeforePreviousTerminator(node, limit) {
+    // Make *absolutely sure* we can't just recurse forever.
+    if (limit == null) {
+      limit = 128;
+    }
+
+    if (limit == 0) {
+      return false;
+    }
+
+    if (node.toString() == ';') {
+      return false;
+    }
+
+    if (node.precedingComments != null &&
+        node.precedingComments.value().contains(ignoreStatement)) {
+      return true;
+    } else {
+      return isIgnoreCommentBeforePreviousTerminator(node.previous, limit - 1);
+    }
+  }
+
+  bool isLineIgnored(node) {
```

Comment:
It seems superficially like this should work for comments within the statement, as with the below example I tried in publishing_client. But it doesn't seem to. If it's hard, we might reasonably leave it for a follow-on PR.
```
 return (Dom.div())(
      // ignore_statement: intl_message_migration
      (Dom.p())(
        'This template will appear in a new ',
        (Dom.span()..className = 'text-bold')(_destinationFolderName()),
        ' folder, which only you can use unless you add collaborators.',
      ),
      (Dom.p())(
        'The template will install in the background while you work; we\'ll let you know when it\'s ready!',
      ),
    );
```

Reply:
@robbecker-wf and I just experimented with that exact case, actually. It doesn't work, but I agree that it absolutely should. We decided that we'll merge this with a known issue and then add a new ticket to address the shortcomings of this version.